### PR TITLE
Introduction of --hwinfo parameter at the cmd line.

### PR DESCRIPTION
**Pull request by fermosan**

### CODE REVIEW

File: src/chia_plot.cpp [hunk 1]
```diff
@@ -178,18 +184,40 @@ int main(int argc, char** argv)
...
 		std::cout << options.help({""}) << std::endl;
 		return 0;
 	}
 	const auto args = options.parse(argc, argv);
+
+	if(maxThreads == cinfo.logicalCpus() * 2) 
+		mPier=2;
+        else if(maxThreads == cinfo.logicalCpus() * 4)
+		mPier=4;
+	maxCores=cinfo.cores() * mPier;
```
Comment (madMAx43v3r) [inline src/chia_plot.cpp:200]: what's the logic behind `mPier`? I don't get it..

Comment (fermosan) [reply src/chia_plot.cpp:200]: The thread is running on 1 core at a time and the ASM calls will give info about the running CPU. There is no easy cross platform way to know all the CPUs on the system. So we take advantage of the std::thread:hardware_concurrency call which will report the total logicalprocessors on the system. Having that information and knowing the setup of one physical CPU we can deduct setups of dual and quad processor systems taking in mind that the setup is symmetric.  mPier is the multiplier of the single CPU cores according to the calculation above.

Comment (madMAx43v3r) [reply src/chia_plot.cpp:200]: so `mPier` == number of CPUs?

Comment (fermosan) [reply src/chia_plot.cpp:200]: Exactly.

File: src/chia_plot.cpp [hunk 2]
```diff
@@ -185,18 +191,40 @@ int main(int argc, char** argv)
...
 	
 	if(args.count("help")) {
 		std::cout << options.help({""}) << std::endl;
 		return 0;
 	}
+	 if (args.count("hwinfo")) {
+                std::cout << "CPU Info          : " << cinfo.model()  <<  " (" << cinfo.vendor() << ")" << std::endl;
+                std::cout << "Logical CPUs      : " << cinfo.logicalCpus() * mPier << std::endl;
+                std::cout << "Core CPUs         : " << cinfo.cores() * mPier << std::endl;
+                std::cout << "HyperThreading    : " << (cinfo.isHyperThreaded() ? "true" : "false") << std::endl;
```
Comment (Feromrk) [inline src/chia_plot.cpp:217]: IMHO http://www.cplusplus.com/reference/ios/boolalpha/ is mor elegant

**Pull request closed without merging**